### PR TITLE
feat: project pipeline create name and source pre check

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
The project pipeline pre checks whether the name is unique under the project and whether the source is created repeatedly

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOls4ODMsNzcyXSwibWVtYmVyIjpbIjEwMDA1NjAiXX0%3D&id=278270&iterationID=772&pId=0&type=TASK)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |        Some interfaces for uniqueness verification of project pipeline      |
| 🇨🇳 中文    |        项目流水线一些唯一性校验的接口      |

### CODE REVIEW

File: modules/dop/providers/projectpipeline/provider.go [hunk 1]
```diff
@@ -36,11 +37,12 @@ type config struct {
...
-	Log      logs.Logger
-	bundle   *bundle.Bundle
-	DB       *gorm.DB           `autowired:"mysql-client"`
-	Register transport.Register `autowired:"service-register" required:"true"`
+	Cfg       *config
+	Log       logs.Logger
+	bundle    *bundle.Bundle
+	DB        *gorm.DB           `autowired:"mysql-client"`
+	Register  transport.Register `autowired:"service-register" required:"true"`
+	APIMTrans i18n.Translator    `translator:"project-pipeline" required:"true"`
```
Comment (Effet) [inline modules/dop/providers/projectpipeline/provider.go]: just `Trans`, not APIMTrans

File: api/proto/dop/projectpipeline/projectpipeline.proto [hunk 2]
```diff
@@ -46,6 +46,49 @@ service ProjectPipelineService {
...
+
+  rpc CreateSourcePreCheck(CreateProjectPipelineSourcePreCheckRequest)
+          returns(CreateProjectPipelineSourcePreCheckResponse) {
+    option(google.api.http) = {
+      get : "/api/project-pipeline/actions/source-pre-check",
+    };
+    option(erda.common.openapi) = {
+      path : "/api/project-pipeline/actions/source-pre-check",
+    };
+  }
```
Comment (Effet) [inline api/proto/dop/projectpipeline/projectpipeline.proto:68]: as a rpc to provide, I think you can combine these two apis to one API like:
`POST /api/project-pipeline/actions/check-create-param`

and the body:
```json
{
  "fields": [
    {
      "field": "name",
      "value": "my name",
    },
    {
      "field": "source",
      "value": "my source",
    }
  ]
}
```

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/providers/projectpipeline/service_impl.go [hunk 3]
```diff
@@ -197,12 +287,22 @@ func (p *ProjectPipelineService) Create(ctx context.Context, params *pb.CreatePr
...
 
-func (p *ProjectPipelineService) checkDefinitionRemoteSameName(appID uint64, name string, userID string) (bool, error) {
-	location, err := p.makeLocationByAppID(appID)
+func (p *ProjectPipelineService) checkDefinitionRemoteSameName(projectID uint64, name string, userID string) (bool, error) {
+	projectDto, err := p.bundle.GetProject(projectID)
+	if err != nil {
+		return false, err
+	}
+
+	orgDto, err := p.bundle.GetOrg(projectDto.OrgID)
```
Comment (sfwn) [inline modules/dop/providers/projectpipeline/service_impl.go:296]: bad performance.
you can add orgname to previous GetProject bundle.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by Effet — APPROVED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**